pagefaultgames/pokerogue
# Primal weather no longer persists if last mon dies to indirect damage

## What are the changes?

Fixes a bug where primal weather would remain active if the last mon in a trainer battle with a primal weather ability (so basically just wave 195 rival's Mega Ray) died to indirect damage.

## Why am I doing these changes?
From Discord issue: https://discord.com/channels/1125469663833370665/1271420013567283220

## What did change?

Previously, if wave 195 rival's Mega Ray died to indirect damage as the last mon in the party, Delta Stream would never get deactivated because the call that applied post faint attributes only actually gets called when the mon is fainted by an attack. This PR adds an else condition to that call that covers indirect damage (in addition to changing other post-faint attributes to account for the possibility of dying to indirect damage).

### Screenshots/Videos



## How to test the changes?




Used overrides.ts to put myself at wave 195 and give myself a mon with an attacking move, Whirlwind, Stealth Rock, and False Swipe. Preferably a Dragon type to ensure Mega Ray comes out before the last slot without having to whirlwind. Once Mega Ray was out, I would False Swipe it to 1 HP, Whirlwind it out, then kill other mons until there was only one mon left besides Mega Ray. Set up Rocks on that mon, kill it, and let Mega Ray come in and die to rocks to observe that Delta Stream correctly gets deactivated.

I'm unsure if this change would warrant an automated test, so plz lmk if I should add one.

## Checklist
